src/parser/expression.js

```diff
@@ -30,26 +30,13 @@ const pp = Parser.prototype;
 // strict mode, init properties are also not allowed to be repeated.
 
 pp.checkPropClash = function (prop, propHash) {
-  if (prop.computed) return;
+  if (prop.computed || prop.kind) return;
 
   const key = prop.key;
-  let name;
-  switch (key.type) {
-    case "Identifier":
-      name = key.name;
-      break;
-
-    case "StringLiteral":
-    case "NumericLiteral":
-      name = String(key.value);
-      break;
-
-    // istanbul ignore next: non-computed property keys are always one of the above
-    default:
-      return;
-  }
+  // It is either an Identifier or a String/NumericLiteral
+  const name = key.type === "Identifier" ? key.name : String(key.value);
```

Comment:
Instead of doing a switch case with two variant just do this one-liner. It will anyway only be Identifier or Literal.